tools/proofers/preview.js

```diff
@@ -895,12 +956,42 @@ $(function () {
             if (viewMode !== "show_tags") {
                 colorString = colorString0 + '>$1</span>';
             }
-            // sub- and super-scripts
-            txt = txt.replace(/_\{([^\}]+)\}/g, '<span class="sub"' + colorString);
-            txt = txt.replace(/\^\{([^\}]+)\}/g, '<span class="sup"' + colorString);
-            // single char superscript -  any char except {
-            // do not allow < incase it's a tag which would screw up
-            txt = txt.replace(/\^([^\{<])/g, '<span class="sup"' + colorString);
+
+            // process sub- and super-scripts
+            function ssProcess(text) {
+                text = text.replace(/_\{(.+?)\}/g, '<span class="sub"' + colorString);
+                text = text.replace(/\^\{(.+?)\}/g, '<span class="sup"' + colorString);
+                // single char superscript -  any char except {
+                // do not allow < incase it's a tag which would screw up
+                return text.replace(/\^([^{<])/g, '<span class="sup"' + colorString);
+            }
+
+            // processes the text by textFunction but in math mode
+            // only outside math markup
+            function processExMath(text, textFunction) {
+                if(!styler.allowMathPreview) {
+                    return textFunction(txt);
+                } else {
+                    // find whole math strings \[ ... \] or \( ... \)
+                    let txtOut = "";
+                    let mathRegex = /\\\[[^]*?\\\]|\\\([^]*?\\\)/g;
+                    for(;;) {
```

Comment:
same here for infinite loops. And if we do use the, we should probably be consistent on for(;;) vs while(true)

Reply:
Yes I think eslint 'no constant condition' will catch while(true) but allow for(;;).
I'll make another branch to tidy this up.